shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java

```diff
@@ -176,12 +180,17 @@ private void unregister(final ApiDO apiDO) {
                     .collect(Collectors.toList()));
         }
         //clean selector
-        Optional.ofNullable(selectorService.findByName(apiDO.getContextPath()))
+        Optional.ofNullable(selectorService.findByNameAndPluginName(apiDO.getContextPath(), PluginEnum.DIVIDE.getName()))
```

Comment:
Other rpc types need to be considered.

Reply:
All proxy plugin IDs should be checked here. In order to quickly fix errors, I only specified the divide plugin. I plan to provide a method to obtain the IDs of all proxy type plugins in the plugin's enumeration class.
But I found two enumeration classes about plugin plugins.  `org.apache.shenyu.e2e.model.Plugin` and `org. apache. shenyu. common. enums. PluginEnum `.
But only the former has a plugin ID. I want to merge `org. apache. shenyu. e2e. model. Plugin` and `org. apache. shenyu. common. enums. PluginEnum`, but the scope of the changes is a bit large, so I want to keep it until the next refactoring. If this PR considers other proxy plugins, I can make minor changes.

Reply:
> Other rpc types need to be considered.

I have already implemented it, push pr immediately.

Reply:
Hi, please review if it's okay. @dragon-zhang